### PR TITLE
mark empty newlines

### PR DESCRIPTION
**Pull request by stevemao**
color the brick (first white space) of each line that is reserved for "+" and "-" that's removed by this script
Alternative of and inspired by #11

diff-so-fancy

### CODE REVIEW

**Comment by paulirish** [comment]
tried out this patch. I'm seeing the inversion applying to text in some cases:

image

**Comment by stevemao** [comment]
@paulirish just pushed the fix. On my Mac I need to close the ansi but not on my Windows...

**Comment by stevemao** [comment]
Oh no there is still some problem :(

**Comment by paulirish** [comment]
Yup, I just saw those. :)   

Those issues fixed, though I found this: 

image

**Comment by noscript** [comment]
You reset the color too early. That should  fix it:
```diff
-	sed -E "s/^($color_code_regex)[\+\-]($reset_color)\s*$/$invert_color\2 \1/g"
+	sed -E "s/^($color_code_regex)[\+\-]($reset_color)\s*$/$invert_color\1 \2/g"
```

**Comment by stevemao** [comment]
I have a feeling, we might be using different versions of `sed`. I updated it on my mac and it works fine (with all the patch suggestions here).

My sed version is 4.2.2

**Comment by noscript** [comment]
Mine is sed (GNU sed) 4.2.2

**Comment by stevemao** [comment]
The `\2` or `\1` here gives me the same results. So I could just do:

```
sed -E "s/^($color_code_regex)[\+\-]$reset_color\s*$/$invert_color\1 \1/g"
```

**Comment by stevemao** [comment]
I think we should always test against the latest sed (GNU sed)

Can you guys verify these patches work for you (all the commits here should work)? I don't really understand why they all work I just blindly tested. It's hard to test these invisible ansi escapes. Can't wait until we have tests (#32)

File: diff-so-fancy [hunk 1]
```diff
@@ -22,9 +23,13 @@ colorize_context_line () {
...
 }
 
+mark_empty_lines () {
+	sed -E "s/^($color_code_regex)[\+\-]$reset_color\s*$/$invert_color\1 /g"
+}
+
 strip_leading_symbols () {
 	# strip the + and -
-	sed -E "s/^($color_code_regex)[\+\-]/\1 /g"
+	sed -E "s/^($color_code_regex)[\+\-]/\1/g"
```
Comment (stevemao) [inline diff-so-fancy]: This change is not needed

**Comment by noscript** [comment]
> The `\2` or` \1` here interchangeable for me the same results.

Even giving the same result, these are not interchangeable really.

**Comment by stevemao** [comment]
The reason is the ansi escape character is captured twice...

**Comment by stevemao** [comment]
@paulirish I understand now. 
It's because your using ` git config --global core.pager "diff-highlight | diff-so-fancy | less -r"`
if you use `git config --global core.pager "diff-so-fancy | less --tabs=1,5 -R"` (the one on readme) it looks fine.

**Comment by stevemao** [comment]
Hope my last commit fixes it.

**Comment by elyscape** [comment]
The screenshot in the readme should probably be updated to reflect this.

**Comment by stevemao** [comment]
@elyscape Just send a PR

**Comment by elyscape** [comment]
@stevemao I can't figure out how the screenshot was hosted, otherwise I would.

**Pull request merged**